compiler/damlc/daml-doc/src/DA/Daml/Doc/Extract.hs

```diff
@@ -330,6 +329,34 @@ getClsDocs ctx@DocCtx{..} (DeclData (L _ (TyClD _ c@ClassDecl{..})) docs) = do
       | L loc name <- names
       ]
     f _ _ = []
+
+    -- | Remove the implied context from typeclass member functions.
+    dropMemberContext :: Maybe Anchor -> [T.Text] -> FunctionDoc -> FunctionDoc
+    dropMemberContext cl_anchor cl_args fn = fn
+        { fct_context = do
+            context <- fct_context fn
+            case context of
+                TypeTuple xs -> do
+                    let xs' = filter (not . matchesMemberContext cl_anchor cl_args) xs
+                    guard (notNull xs')
+                    Just (TypeTuple xs')
+
+                _ -> Just context
+                    -- TODO: Move to using a more appropriate type
+                    -- for contexts in damldocs, to avoid this case.
+        }
+
+    -- | Is this the implied context for member functions? We use an anchor
+    -- for comparison because it is more accurate than typenames (which are
+    -- generally susceptible to qualification and shadowing).
+    matchesMemberContext :: Maybe Anchor -> [T.Text] -> DDoc.Type -> Bool
+    matchesMemberContext cl_anchor cl_args ty = and
+        [ cl_anchor == getTypeAppAnchor ty
+        , Just [(Just (Typename arg), Just []) | arg <- cl_args] ==
```

Comment:
I guess we can’t use an anchor here as well since there is no easy way to get access to one?

Reply:
These are type variables so there isn't an anchor.

Reply:
Makes sense, thanks for the explanation.